tasks/office.py

```diff
@@ -66,34 +73,16 @@ def add_stored(self, object_key, object_data):
 
         if len(data_list) >= self.DATA_LIST_MAX:
             num_to_delete = random.randint(1, len(data_list))
-            del data_list[:num_to_delete]
+            # Convert a dict to a list so we can take a slice by insertion order fifo
+            self.local_store[object_key] = dict(list(data_list.items())[num_to_delete:])
```

Comment:
This is an interesting edit! We might want to do `num_to_delete - 1` because this would cause an exception if it happened to be the value of the length of the list (that's my bad originally).